### PR TITLE
Make a function in 'type_case.h' static inline

### PR DESCRIPTION
**Pull request by theartful**

### CODE REVIEW

File: src/shogun/lib/type_case.h [hunk 1]
```diff
@@ -145,7 +145,7 @@ namespace shogun
 #undef SG_ADD_SGVECTOR_TYPE
 #undef SG_ADD_SGMATRIX_TYPE
 
-		std::string print_map(const typemap& map)
+		static inline std::string print_map(const typemap& map)
```
Comment (karlnapf) [inline src/shogun/lib/type_case.h]: if you want to inline things use `SG_FORCE_INLINE`

**Comment by karlnapf** [comment]
thanks for this!

@gf712 have a look as well

File: src/shogun/lib/type_case.h (hunk 1 above)
Comment (gf712) [reply src/shogun/lib/type_case.h]: this is an inline specifier for the linker, the code itself doesn't have to be inlined, so don't have to force it

**Comment by gf712** [comment]
> thanks for this!
> 
> @gf712 have a look as well

Yup, this is the issue that @geektoni had in his PR

File: src/shogun/lib/type_case.h (hunk 1 above)
Comment (karlnapf) [reply src/shogun/lib/type_case.h]: I see thanks!

**Comment by karlnapf** [comment]
Ill wait for CI and then merge if there are no complaints

File: src/shogun/lib/type_case.h (hunk 1 above)
Comment (vigsterkr) [reply src/shogun/lib/type_case.h]: `SG_FORCED_INLINE` plz

File: src/shogun/lib/type_case.h [hunk 2]
```diff
@@ -158,7 +158,7 @@ namespace shogun
 			return msg;
 		}
 
-		TYPE get_type(const Any& any, const typemap& map)
+		static inline TYPE get_type(const Any& any, const typemap& map)
```
Comment (vigsterkr) [inline src/shogun/lib/type_case.h]: same here...

File: src/shogun/lib/type_case.h (hunk 1 above)
Comment (theartful) [reply src/shogun/lib/type_case.h]: @vigsterkr this is to satisfy the one definition rule for the linker. should it be `SG_FORCED_INLINE`ed?

Comment (vigsterkr) [reply src/shogun/lib/type_case.h]: yes plz force it

**Comment by gf712** [comment]
@karlnapf this should be good to go

**Comment by geektoni** [comment]
What's the status of this one? Because I would really love to have it merged soon :P

**Pull request merged**